File: src/main/java/io/github/takenoha/towerdefense/paper/RaidSealRecipeDefinition.java
package io.github.takenoha.towerdefense.paper;

import java.util.List;

/** Vanilla ingredients shared by every stage-specific raid-seal recipe. */
public final class RaidSealRecipeDefinition {
    public static final String PAPER_MATERIAL = "PAPER";

    private RaidSealRecipeDefinition() {
    }

    public static List<String> shape() {
        return List.of(" P ", "PSP", " P ");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/paper/RaidSealRecipeDefinition.kt
package io.github.takenoha.towerdefense.paper

/** Vanilla ingredients shared by every stage-specific raid-seal recipe. */
class RaidSealRecipeDefinition private constructor() {
    companion object {
        const val PAPER_MATERIAL: String = "PAPER"

        @JvmStatic
        fun shape(): List<String> = listOf(" P ", "PSP", " P ")
    }
}
